docroot/profiles/custom/sitenow/sitenow.profile

```diff
@@ -515,6 +515,28 @@ function sitenow_preprocess_page(&$variables) {
   if (!$admin_context->isAdminRoute()) {
     $node = \Drupal::routeMatch()->getParameter('node');
     $node = (isset($node) ? $node : \Drupal::routeMatch()->getParameter('node_preview'));
+    if (isset($node)) {
+      // Get moderation state of node.
+      $revision_id = $node->getRevisionId();
+      $revision = \Drupal::entityTypeManager()->getStorage('node')->loadRevision($revision_id);
+      $moderation_state = $revision->get('moderation_state')->getString();
+      switch ($moderation_state) {
```

Comment:
Can I propose an alternative:
```
      // If not published (status = 1), display a message indicating the current workflow state.
      $status = $revision->get('status')->value;
      if ($status == 0) {
        $warning_text = t('This content is currently in a <em>"@moderation_state"</em> state.', [
          '@moderation_state' => $moderation_state,
        ]);
        \Drupal::messenger()->addWarning($warning_text);
      }
```

Reply:
This keeps the language consistent across states, less lines and allows for additional moderation states in the future. Even moderation states that transition the node to status 1.

Reply:
Updated to check on status rather than moderation_state, and updated language (I see now there's a nice example right above this in _sitenow_prevent_front_delete_message...)

Made a minor tweak: seeing '...a "archived"...' made my skin itch. Adds two extra lines, but keeps in line with allowing additional moderation states in the future without requiring updates.

Reply:
haha, i didn't think about that. creative workaround.